src/midictl.c

```diff
@@ -482,13 +482,50 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 
+	bool save_config_path = 1;
+	// check if config file path is given
+	//if not open the last given path.
+	if (config.config_path == NULL)
+	{
+		FILE *prev_config_path = fopen(".prevconfpath", "rt");
+		if (prev_config_path == NULL)
+		{
+			//prev file doesnt exist
+			fprintf(stderr, "Previously used config path is non existant\n");
+			exit(EXIT_FAILURE);
+		}
+		
+		//get the file size
+		fseek(prev_config_path, 0L, SEEK_END);
+    	uint32_t sz = ftell(prev_config_path) + 1;
+    	fseek(prev_config_path, 0L, SEEK_SET);
+
+		//get the path
+		char* f_path = malloc(sz);
```

Comment:
Hardly significant, but it seems that you missed `free()` for this allocation. I guess the simplest solution here is to change `config.config_path` to `char*` and always store a dynamically allocated string there and always free it at the end.

Note that this code would need to use `strdup()` then:
https://github.com/Jacajack/midictl/blob/c0f11ac34ce451d4d6511eb426a1609cc6edb948/src/args.c#L37

Reply:
Admittedly this is my first time using malloc(). I went a different direction than what you said here. I'm also still new to pull requests. I've committed the change I made, but do I need to open a new pull request for you to be able to review? Sorry for my lack of github experience ;)

Reply:
It's fine, I can see your changes here :)